HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift

```diff
@@ -114,47 +114,97 @@ extension HILoginFlowController {
 
 // MARK: - Login Flow
 extension HILoginFlowController {
-    func populateUserData(loginMethod: HILoginMethod, token: String, sender: HIBaseViewController) {
-        HIUserService.getUser(by: token, with: loginMethod)
-        .onCompletion { result in
-            switch result {
-            case .success(let containedUser, _):
-                let userInfo = containedUser.data[0]
-                var user = HIUser(
-                    loginMethod: loginMethod,
-                    permissions: userInfo.roles.map { $0.permissions }.reduce(.guest, max),
-                    token: token,
-                    identifier: userInfo.info.email,
-                    isActive: true,
-                    id: userInfo.info.id,
-                    name: nil,
-                    dietaryRestrictions: nil
-                )
-
-                HIRegistrationService.getAttendee(by: token, with: loginMethod)
-                .onCompletion { result in
-                    switch result {
-                    case .success(let containedAttendee, _):
-                        let attendeeInfo = containedAttendee.data[0]
-                        let names = [attendeeInfo.firstName, attendeeInfo.lastName].compactMap { $0 } as [String]
-                        user.name = names.joined(separator: " ")
-                        user.dietaryRestrictions = attendeeInfo.diet
-
-                    case .failure:
-                        break
-                    }
-
+    func getUserTokenFromCode(code: String) {
+        let json: [String: Any] = ["code": code]
```

Comment:
This should be an APIManager endpoint, there are very few reasons to use raw URLSessionDataTasks as it hurts maintainability.

Reply:
Do you mean something like this?

```
static func create(title: String, description: String) -> APIRequest<HIAPIAnnouncement.Contained> {
        var announcementDict = [String: Any]()
        announcementDict["title"]       = title
        announcementDict["description"] = description
        return APIRequest<HIAPIAnnouncement.Contained>(service: self, endpoint: "", body: announcementDict, method: .POST)
    }
```

Reply:
Yeah